java/impl/gcp/src/main/java/co/worklytics/psoxy/GcpModule.java

```diff
@@ -61,6 +58,38 @@ static LockService lockService(@Named("instance") SecretManagerConfigService ins
         return instanceConfigService;
     }
 
+    @Provides @Singleton
+    static SecretStore secretStore(@Named("instance") SecretManagerConfigService instanceConfigService,
+                                   HostEnvironment hostEnvironment,
+                                   VaultConfigServiceFactory vaultConfigServiceFactory,
+                                   EnvVarsConfigService envVarsConfigService) {
+        if (vaultConfigServiceFactory.isVaultConfigured(envVarsConfigService)) {
+            if (!vaultConfigServiceFactory.isVaultConfigured(envVarsConfigService)) {
+                throw new IllegalStateException("Vault is not configured, but HASHICORP_VAULT secret store implementation was requested");
+            }
+            VaultConfigService sharedVault =
+                vaultConfigServiceFactory.create(VaultConfigServiceFactory.pathForSharedVault(hostEnvironment, envVarsConfigService));
+
+            VaultConfigService connectorVault =
+                vaultConfigServiceFactory.create(VaultConfigServiceFactory.pathForInstanceVault(hostEnvironment, envVarsConfigService));
+
+            //initialize vaults (q: right place for this??)
```

Comment:
Same code than AWS, right? If that's the case, anything that we could do for reusing it? Maybe create a `VaultFactory` or something for that?

Reply:
thanks, refactored a bit.